### PR TITLE
Add the weekly percent change above vitals chart

### LINKED ISSUE #945: Add 7 and 28 day change for the vitals over time chart
**Is your feature request related to a problem? Please describe.**
Add the 7-day change and 28-day change components above the vitals over time chart.

image

**Describe the solution you'd like**
N/A
**Describe alternatives you've considered**
N/A
**Additional context**
Original chart issue: #943
Mocks: https://www.figma.com/file/H08AQJSaAIRxfeMP4iCUPh/UP-MVP?node-id=2022%3A4

### PR DESCRIPTION
**Pull request by daschi**
## Description of the change

This PR adds the 7-day and 28-day percent change above the Vitals chart. These numbers are calculated from the time series data. 

Screen Shot 2021-03-26 at 9 33 36 AM

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes #945 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/core/controls/PercentDelta.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+const deltaColorMap: {
+  [key in DeltaDirections]: string;
+} = {
+  improved: styles.signalLinks,
+  worsened: styles.crimsonDark,
+  noChange: styles.slate60,
+} as const;
+
+type DeltaDirections = keyof typeof deltaDirections;
```
Comment (Catacola) [inline src/core/controls/PercentDelta.tsx:41]: Genius

File: src/core/PageVitals/index.tsx [hunk 2]
```diff
@@ -20,26 +20,65 @@ import { observer } from "mobx-react-lite";
...
+  { date: "2021-03-22", value: 67, weeklyAvg: 54, parentWeeklyAvg: 36 },
+  { date: "2021-03-23", value: 79, weeklyAvg: 97, parentWeeklyAvg: 20 },
+  { date: "2021-03-24", value: 69, weeklyAvg: 94, parentWeeklyAvg: 87 },
+  { date: "2021-03-25", value: 22, weeklyAvg: 75, parentWeeklyAvg: 13 },
+  { date: "2021-03-26", value: 14, weeklyAvg: 50, parentWeeklyAvg: 76 },
+  { date: "2021-03-27", value: 3, weeklyAvg: 59, parentWeeklyAvg: 59 },
+  { date: "2021-03-28", value: 83, weeklyAvg: 85, parentWeeklyAvg: 38 },
+];
+
+const twentyEightDaysAgo = mockTimeSeriesData[0];
```
Comment (jovergaag) [inline src/core/PageVitals/index.tsx:75]: Are we certain that the earliest date will always be first? Should we sort it first to be sure?

**Review by Catacola — APPROVED** [review]

File: src/core/controls/PercentDelta.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+type PropTypes = {
+  value: number;
+  className?: string;
+  width?: number;
+  height?: number;
+  improvesOnIncrease: boolean;
+};
+
+const PercentDelta: React.FC<PropTypes> = ({
```
Comment (jovergaag) [inline src/core/controls/PercentDelta.tsx:92]: It would be nice to have a test for this considering the logic in `getDeltaDirection`

**Review by jovergaag — CHANGES_REQUESTED** [review]

File: src/core/PageVitals/index.tsx (hunk 2 above)
Comment (daschi) [reply src/core/PageVitals/index.tsx:75]: I'm not sure how the view is being sorted, but yes we should definitely sort the records by date when they're first transformed from the raw api data.

File: src/core/VitalsWeeklyChange/index.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+          improvesOnIncrease
+        />
+      </div>
+    </div>
+  );
+};
+
+const VitalsWeeklyChange: React.FC<PropTypes> = ({ data }) => {
+  // todo - calculate 7day and 28day values from timeseries data
+  // todo - send a classname to percentdelta to update styles
```
Comment (jovergaag) [inline src/core/VitalsWeeklyChange/index.tsx]: Did you want to wrap up these TODOs in this PR or is there one that follows?

**Review by jovergaag — APPROVED** [review]
🔽 ⏫

File: src/core/VitalsWeeklyChange/index.tsx (hunk 4 above)
Comment (jovergaag) [reply src/core/VitalsWeeklyChange/index.tsx]: NVM - was going commit by commit and found them resolved later!

**Pull request merged**